src/index/index.h

```diff
@@ -6,73 +6,41 @@
 
 #pragma once
 
+#include "index_field.h"
+#include "redisearch_api.h"
+#include "../graph/graph.h"
 #include "../graph/entities/node.h"
 #include "../graph/entities/edge.h"
+#include "../filter_tree/filter_tree.h"
 #include "../graph/entities/graph_entity.h"
-#include "../graph/graph.h"
-#include "redisearch_api.h"
 
-#define INDEX_OK 1
-#define INDEX_FAIL 0
+#define INDEX_OK   1  // index operation succeeded
+#define INDEX_FAIL 0  // index operation failed
 #define INDEX_SEPARATOR '\1'  // can't use '\0', RediSearch will terminate on \0
-#define INDEX_FIELD_NONE_INDEXED "NONE_INDEXABLE_FIELDS"
-
-#define INDEX_FIELD_DEFAULT_WEIGHT 1.0
-#define INDEX_FIELD_DEFAULT_NOSTEM false
-#define INDEX_FIELD_DEFAULT_PHONETIC "no"
-
-// initialize index field with default values
-#define IndexField_Default(field, id, name) IndexField_New(field, id, name,  \
-		INDEX_FIELD_DEFAULT_WEIGHT, INDEX_FIELD_DEFAULT_NOSTEM,              \
-		INDEX_FIELD_DEFAULT_PHONETIC)
 
 // forward declaration
 typedef struct _Index _Index;
 typedef _Index *Index;
 
+// deprecated
 typedef enum {
-	IDX_ANY          =  0,
-	IDX_EXACT_MATCH  =  1,
-	IDX_FULLTEXT     =  2,
+	IDX_ANY         = 0,
+	IDX_EXACT_MATCH = 1,
+	IDX_FULLTEXT    = 2,
```

Comment:
is this needed?

Reply:
I believe it is, old encoded RDBs will contain index type.